doc/rst/source/debug.rst

```diff
@@ -0,0 +1,129 @@
+.. index:: ! debug
+
+************************
+Debugging for Developers
+************************
+
+GMT developers may need to run GMT in a debugger to figure out the problem reported by
+an issue.  Often there is a bit of detective work involved to find out where a module
+crashes and then step carefully through that section, examining variables etc., to learn
+why something fails.  This process depends on the particular debug tool one uses.  This page
+will explain the steps a developer must take to build gmt so it is suitable for your debug
+tool and how to use that tool.
+
+Xcode on macOS
+--------------
+
+Developers on macOS are most likely to want to use Xcode for this purpose. Chances are they
+are already using the command-line tools from Xcode (gcc, linker) anyway.  To use the GUI
+part of Xcode you must have installed the full Xcode (you most likely did, but there are
+ways to *only* install the command-line tools, so make sure you have an Xcode icon under
+Applications).  Xcode may change as versions change; the images below is for Xcode 10-11.
+
+#. You will need to make some changes to your *cmake/ConfigUser.cmake* file. Scroll down to the
+   section that says "# Extra debugging for developers:" and uncomment the ~6 lines that has
+   the if test on "Xcode".  This will pas a few flags that are using when debugging via Xcode.
+   Also uncomment the two "add_definitions" lines that contain the word "DEBUG" in it somewhere.
+
+#. You will need to build GMT again.  I recommend you do this into another build directory (I
+   call mine xbuild, and while at it I use rbuild for "Release Builds" and dbuild for regular
+   debug command line builds, and build for the final build for the releases).  If you are in
+   the xbuild directory, these two command will do the build::
+
+    cmake -DCMAKE_INSTALL_PREFIX=gmt6 -DCMAKE_BUILD_TYPE=Debug -G Xcode ..
+    xcodebuild
+
+#. After this step you can launch the Xcode application and then use the File->Open menu to
+   open the project file xbuild/GMT.xcodeproj.  After it opens it may looks something like
+   this:
+
+.. figure:: /_images/xcode-1.*
+   :width: 500 px
+   :align: center
```

Comment:
```
.. figure:: /_images/xcode-1.*
   :width: 500 px
   :align: center
```
@PaulWessel Indent the whole figure directive can fix your formatting issue 1.

Reply:
Thanks, works fine.  Added last figure before heading home.  This takes us through all the steps to get to top of a given module.  The rest now depends on the bug.